src/Controls/src/Build.Tasks/XmlTypeExtensions.cs

```diff
@@ -18,17 +20,12 @@ static IList<XmlnsDefinitionAttribute> GatherXmlnsDefinitionAttributes(ModuleDef
 			{
 				// Search for the attribute in the assemblies being
 				// referenced.
+				GatherXmlnsDefinitionAttributes(xmlnsDefinitions, module.Assembly);
+
 				foreach (var asmRef in module.AssemblyReferences)
 				{
 					var asmDef = module.AssemblyResolver.Resolve(asmRef);
-					foreach (var ca in asmDef.CustomAttributes)
-					{
-						if (ca.AttributeType.FullName == typeof(XmlnsDefinitionAttribute).FullName)
-						{
-							var attr = GetXmlnsDefinition(ca, asmDef);
-							xmlnsDefinitions.Add(attr);
-						}
-					}
+					GatherXmlnsDefinitionAttributes(xmlnsDefinitions, asmDef);
```

Comment:
should only collect XmlnsDefinition for global namespaces from the local assemby